File: 자료구조/백준_10828_스택/스택.py
from sys import stdin

n = int(stdin.readline())
arr = [0 for _ in range(10001)]
f = 0

while n > 0:
    n-=1
    str = list(stdin.readline().split())
    if str[0] == "push":
        arr[f] = int(str[1])
        f+=1
    elif str[0] == "pop":
        if f == 0: print("-1")
        else:
            print(arr[f-1])
            arr[f-1] = 0
            f-=1
    elif str[0] == "size": print(f)
    elif str[0] == "empty":
        if f == 0: print("1")
        else: print("0")
    elif str[0] == "top":
        if f == 0 : print("-1")
        else: print(arr[f-1])

File: 자료구조/백준_10828_스택/스택.java
import java.io.*;
import java.util.*;

public class Main{
    public static void main(String[] args)throws IOException{
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Scanner s = new Scanner(System.in);
        int n = s.nextInt();
        int[] arr = new int[10001];
        int f = 0;
        while(n-- > 0){
            String str = s.next();
            if(str.equals("push")){
                int num = s.nextInt();
                arr[f] = num;
                f++;
            }
            else if(str.equals("pop")){
                if(f == 0) {
                    bw.write("-1");
                    bw.newLine();
                }
                else{
                    bw.write(Integer.toString(arr[f-1]));
                    bw.newLine();
                    arr[f - 1] = 0;
                    f--;
                }
            }
            else if(str.equals("size")){
                bw.write(Integer.toString(f));
                bw.newLine();
            }
            else if(str.equals("empty")){
                if(f == 0) {
                    bw.write("1");
                    bw.newLine();
                }
                else{
                    bw.write("0");
                    bw.newLine();
                }
            }
            else if(str.equals("top")){
                if(f == 0){
                    bw.write("-1");
                    bw.newLine();
                }
                else{
                    bw.write(Integer.toString(arr[f-1]));
                    bw.newLine();
                }
            }
        }
        bw.flush();
        bw.close();
    }
}